### PR TITLE
feat(api): add plugin allowing server to update cookies

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
This plugin allows the server to give new attributes to existing cookies

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: api/src/plugins/cookie-update.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  options,
+  done
+) => {
+  fastify.addHook('onSend', (request, reply, payload, next) => {
+    for (const cookie of options.cookies) {
+      const oldCookie = request.cookies[cookie];
+      if (!oldCookie) continue;
+
+      const unsigned = unsign(oldCookie);
+      const raw = unsigned.valid ? (unsigned.value as string) : oldCookie;
```
Comment (naomi-lgbt) [inline api/src/plugins/cookie-update.ts:30]: Can we not type the unsign function better?

Something like `{ valid: false, value: null } | { valid: true, value: string }`?

Comment (ojeytonwilliams) [reply api/src/plugins/cookie-update.ts:30]: Yeah, we could do that, but I went down the rabbit hole a little and created a PR to fastify cookie: https://github.com/fastify/fastify-cookie/pull/292

If they're happy with that, we'll get an eslint error when we update and can just drop `as string` at that point.

**Comment by ojeytonwilliams** [comment]
Just to make sure there were no false positive tests, I rebased this on top of https://github.com/freeCodeCamp/freeCodeCamp/pull/55425 

If that goes in first, I'll clean this up.

**Review by naomi-lgbt — APPROVED** [review]
So much cleaner!

**Review by Sembauke — APPROVED** [review]

**Pull request merged**